src/main/java/net/fabricmc/loom/AbstractPlugin.java

```diff
@@ -325,12 +326,15 @@ protected void configureCompile() {
 
 				try {
 					AbstractArchiveTask sourcesTask = (AbstractArchiveTask) project1.getTasks().getByName("sourcesJar");
+					String oldClassifier = sourcesTask.getClassifier();
+					sourcesTask.setClassifier(Strings.isNullOrEmpty(oldClassifier) ? "dev" : oldClassifier + "-dev");
```

Comment:
is it valid to have a classifer contain `-`?

does it even matter about the old classifer?

Reply:
hmm... this works in `gradle build`
classifier confusion probably will push us closer to artifact transformation...

Reply:
What is the use of handling the old classifer here?

Reply:
adds dev to the old jar's classifier and use the old classifier on the new remapped task's classifer